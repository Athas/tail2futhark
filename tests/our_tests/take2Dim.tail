let v0:[int]2 = reshape{[int],[1,2]}([2,3],iotaV(10)) in
let v1:[int]2 = take{[int],[2]}(0,v0) in
i2d(reduce{[int],[0]}(addi,0,reshape{[int],[2,1]}([3],v1)))
