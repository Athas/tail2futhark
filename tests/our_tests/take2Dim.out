fun real main() =
  let t_v0 = reshape((2,3),reshape1_int((2 * (3 * 1)),reshape(((size(0,iota(10)) * 1)),iota(10)))) in
  let t_v1 = reshape((if (0 <= 0) then -0 else 0,size(1,t_v0)),take1_int((0 * (size(1,t_v0) * 1)),reshape(((size(0,t_v0) * (size(1,t_v0) * 1))),t_v0))) in
  toReal(reduce(+,0,reshape((3),reshape1_int((3 * 1),reshape(((size(0,t_v1) * (size(1,t_v1) * 1))),t_v1)))))